keyboards/jj_ergo/jj_ergo.c

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2019 Neil Kettle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "jj_ergo.h"
+#include QMK_KEYBOARD_H
+
+#ifdef RIGHT_HALF
```

Comment:
I'd like to see as much of the MCP23018 code contained within the custom matrix as possible.

Reply:
I've now moved the offending code into the custom matrix only.